### DISCUSSION TITLE
Module4: cannot run slither on single sol file

### DISCUSSION THREAD

**Discussion by Vincent71399** [Cyfrin/security-and-auditing-full-course-s23#211] [Q&A] [answered]
slither . works fine
but slither src/PuppyRaffle.sol (or other very simple sol file), always give error: 

'forge config --json' running
Could not detect solc version from Foundry config. Falling back to system version...
'solc --version' running
'solc @openzeppelin/contracts=lib/openzeppelin-contracts/contracts/ base64/=lib/base64/ ds-test/=lib/forge-std/lib/ds-test/src/ forge-std/=lib/forge-std/src/ openzeppelin-contracts
/=lib/openzeppelin-contracts/contracts/ src/PuppyRaffle.sol --combined-json abi,ast,bin,bin-runtime,srcmap,srcmap-runtime,userdoc,devdoc,hashes,compact-format --optimize --optimize-runs 200 --evm-version paris --allow-paths .,/mnt/c/Users/lutar/IdeaProjects/4-puppy-raffle-audit/src' running
Compilation warnings/errors on src/PuppyRaffle.sol:
Invalid option for --evm-version: paris

ERROR:Slither:Unable to compile all targets.

**Comment by EngrPips** [comment]
I haven't used Slither so much to run into this error, but in the output you provided, it says `Invalid option for --evm-version: paris` Maybe you can change the `evm-version` in `foundry.toml` and see if that will resolve the issue.

**Comment by cas443** [comment, marked as answer]
When you're trying to run Slither on a single file, it doesn't know which Solidity version should be used. You can manually specify it in _foundry.toml_ file in the root of the project like so and it'll run:

**foundry.toml**
```diff
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
+ solc = '0.7.6'

remappings = ['@openzeppelin/contracts=lib/openzeppelin-contracts/contracts']

# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
```

**Reply by EngrPips** [reply]
That is interesting. Thanks for the heads up, sir. @ Vincent71399, could you try this and let us know how it goes?

**Reply by Vincent71399** [reply]
adding + solc = '0.7.6' works for my side, many thanks

**Reply by EngrPips** [reply]
Awesome, I'm glad you figured it out. Keep crushing it.

**Comment by lopjf** [comment]
What if I already use `auto_detect_solc = true` in the TOML?

**Reply by EngrPips** [reply]
If it works fine then I assume you are good.
